app/assets/javascripts/dashboard.js

```diff
@@ -0,0 +1,56 @@
+function setupTaskItemEvents() {
+  var titleElements = document.querySelectorAll(".task-item .task-title"),
+      caretElements = document.querySelectorAll(".task-item .fa");
+
+  addEventListenerToElements(titleElements, "click", onTaskTitleClick);
+  addEventListenerToElements(caretElements, "click", onTaskCaretClick);
+}
+
+function onTaskTitleClick(e) {
+  var taskItem   = e.target.parentNode,
+      checkboxId = e.target.id || e.target.getAttribute("for"),
+      projectId  = checkboxId.split("-")[0],
+      taskId     = checkboxId.split("-")[1];
+
+      csrfToken = document.querySelector("meta[name=csrf-token]")
+                  .getAttribute("content");
+
+  reqwest({

```

Comment:
you want to do this in the future so you are adding another lib? :P.. How far this future is? I mean this is something in the backlog that will happen in really near future or this is a plan for the future development and you need to discuss this with the team?